### ISSUE TITLE
Scheduler inefficiency and management of local and remote workers

### ISSUE THREAD

**Issue by Meatball13** [filecoin-project/lotus#3410]
**Describe the bug**
Scheduler is being very inefficient about how it manages handing out work to the internal and remote worker.  Often times leaving workers idle for long periods of time while work queues up on other worker.

**To Reproduce**
1. Install Lotus on Main Box, run daemon/miner
2. Install Lotus on remote worker, run lotus-worker
3. lotus-miner sectors pledge 2 times to get two sectors running.

**Expected behavior**
The scheduler be a bit more intelligent about how it handles all the tasks associated with sealing a sector and managing them between workers.  Seems to often leave some workers completely idle for long stretches while work queues up on other work.  Will often transfer data between machines when it would be much faster to stay on the original worker machine.  On a clean setup with two new workers it took a total of 15:43 to complete the two sectors pledge tasks.  Would expect it to take about 8-9 hours per sector, and with two workers running, would expect maybe 10 hours to complete two sectors in parallel. 

**Screenshots**

**Version (run `lotus version`):**
lotus version
Daemon:  0.5.6+git.606a58bc+api0.13.0
Local: lotus version 0.5.6+git.606a58bc

**Additional context**
Attached the related lotus-miner and lotus-worker logs as well as sector logs for two sectors.

Two boxes with similar hardware (Ryzen 3900X, 128GB DDR4, 2 TB NVME Main/Sealing Drive, 1 TB NVME Swap drive) with the only difference is the main Daemon box has a x570 Mothboard and GTX 1070 Ti GPU, external Worker box has a B450 motherboard, GTX 1070 GPU.

One box called 'Daemon' setup with lotus daemon and lotus-miner.  Lotus-Miner set to true for AllowAddPiece, AllowPreCommit1, AllowPrecommit2, AllowCommit, AllowUnseal in config.toml.

One external worker called 'Worker' with lotus installed.  MINER_API_INFO string set with API key from Daemon box and both boxes see the workers on both machines.  Connected over a 1GbE network link.  lotus-worker launched with following command, 'nohup lotus-worker --worker-repo ~/worker run > ~/worker.log 2>&1 &' There is an empty config.toml in worker repo.

Started two sector pledges within 30 minutes of each other.  Daemon box starts working PC1 for sector 1, Worker box starts working PC1 for sector 2.

Daemon begins worker sector 1 PC1 @ 07:25
Worker begins working sector 2 PC1 @ 08:03
Worker completes sector 2 PC1 @ 11:20 (3:54)
Worker completes sector 2 PC2 @ 12:43 (1:23)
Worker completes sector 2 PreCommitted/PrecommitLanded and begins SeedReady Phase @ 12:47 (0:04)
Daemon completes sector 1 PC1 @ 13:26 (5:22)
Daemon attempts to server Sector 1 PC1 data to Worker @ 13:26
Worker finishes sector 2 SeedReady phase @ 14:02
Both machines idle while transfer completes
Worker gets sector 1 PC1 data and begins working on sector 1 PC2 @ 17:48 (4:22 Transfer time...)
Worker completes sector 1 PC2 @ 19:12 (1:24)
Worker completes sector 1 PreCommitted/PrecommitLanded and begins SeedReady Phase @ 19:15 (0:03)
Worker completes sector 1 Committed/CommitSubmitted @ 21:02 (1:47)
Worker serves finalized sector 1 to Daemon @ 21:06 (0:04)
Daemon receives full Sector 1 @ 21:11 and sector is successfully 'Proving' (0:05)
Worker completes sector 2 Committed/CommitSubmitted @ 22:51 (1:39)
Worker serves finalized sector 2 C2 to Daemon @ 22:56 (0:05)
Daemon receives finalized Sector 2 @ 23:08 and sector is successfully 'Proving'

[daemon-lotus-miner.log](https://github.com/filecoin-project/lotus/files/5148259/daemon-lotus-miner.log)
[daemon-lotus-miner.log](https://github.com/filecoin-project/lotus/files/5148260/daemon-lotus-miner.log)
[sector-status-logs.txt](https://github.com/filecoin-project/lotus/files/5148261/sector-status-logs.txt)

**Comment by Meatball13** [comment]
One follow-up.  Once those sectors completed, I started two more pledges and the exact same thing happened.  Once PC1 completed on the Daemon box, it started moving it over to the Worker box.  2 hours in and only 6 of the 11 layer files have been transferred, so I expect about 4 hours again for the transfer.  In the meantime, the sector that was being worked on the worker box is idling/waiting for the transfer (and probably the PC2 work for the incoming sector) to complete.

```
ID  Sector  Worker  Hostname    Task  State        Time
5   3       2       tbthdwork1  GET   running      2h2m44s
5   4       3       TBTHDWORK2  C2    running      1h5m50.8s
```

 I've noticed two additional things as well:

1) The transfer rate between the two boxes is not saturating the link between them.  It seems to burst up occasionally to full speed, but then drop down.  So instead of running at max speed (~110MiB/s in my case) and finishing up probably in 60-90 minutes, it often drops down to < 10 MiB/s and just takes a long time.  Not sure if this is due to other network traffic on the Daemon box (Daemon traffic?) or what.  Certainly not getting near the limits of the NVME read/write speeds so it shouldn't be bottlenecked there.

![netSpeedReceiving](https://user-images.githubusercontent.com/29860922/91715325-678bfc00-eb5b-11ea-8982-e94da80a5d96.jpg)

2) Maybe a simple fix could be some sort of flag that you could set to on the miner/scheduler box to force the scheduler to always keep sector work on the same machine until it's complete and never try to move it between boxes.  Unless you have 10GbE networking, you're probably going to lose more time in the transfer process than you gain by moving jobs around.

**Comment by Meatball13** [comment]
One follow up.  I do think the problems with the network speed were related to an old switch.  Swapped that out and getting better transfer speeds in my initial tests.  Will report back if not.  That being said, still would be nice to have a bit more control over letting stuff transfer data, especially in the PC1 to PC2 handoff since that's such a huge block of data.
